tests/Homework 17/registration_pom.spec.js

```diff
@@ -0,0 +1,293 @@
+import WelcomePage from "../pageObject/WelcomePage.js";
+import SignInPopup from "../components/SignInPopup.js";
+import RegistrationPopup from "../components/RegistrationPopup.js";
+import {expect, test} from "@playwright/test";
+
+test.describe('Registration', async () => {
+    let page
+    let welcomePage
+
+    test.beforeEach(async ({browser}) => {
+        page = await browser.newPage()
+        welcomePage = new WelcomePage(page)
+        await welcomePage.visit()
+
+    })
+
+    test('Registration with correct data', async () => {
+
+        await welcomePage.signInButton.click()
+        const signInPopup = new SignInPopup(page)
+        await signInPopup.registrationButton.click()
+        const registrationPopup = new RegistrationPopup(page)
+
+        await registrationPopup.inputName.fill('Olena')
+        await registrationPopup.inputLastName.fill('Nikitina')
+        await registrationPopup.inputEmail.fill(registrationPopup.generateRandomEmail())
+        await registrationPopup.inputPassword.fill('1321@Helena')
+        await registrationPopup.inputPasswordConfirmation.fill('1321@Helena')
+        await registrationPopup.registerButton.click()
+    })
+
+    test('Name validation - number of symbols', async () => {
+
+        await welcomePage.signInButton.click()
+        const signInPopup = new SignInPopup(page)
+        await signInPopup.registrationButton.click()
+        const registrationPopup = new RegistrationPopup(page)
```

Comment:
this actions are repeating every test, move it to beforeEach

Reply:
by the way there is single sign up button you can use
![image](https://github.com/onikitina1321/aqa-playwright/assets/53534240/b897cd87-034b-4b92-92b1-d9e97dba6975)